net-vpn/i2pd: fix build with libressl #597798

khumarahn opened the pull request:
i2pd had a building problem with libressl:
https://bugs.gentoo.org/597798
https://github.com/PurpleI2P/i2pd/issues/686

The issue was a boost/asio problem, which has been fixed in boost-1.65. This PR adds a requirement for >=boost-1.65 in case of using libressl.

blueness commented on net-vpn/i2pd/i2pd-2.13.0.ebuild:
```diff
@@ -12,15 +12,16 @@ SLOT="0"
+RDEPEND="!static? ( dev-libs/boost[threads]
 			!libressl? ( dev-libs/openssl:0[-bindist] )
-			libressl? ( dev-libs/libressl )
+			libressl? ( dev-libs/libressl >=dev-libs/boost-1.65 )
```
Can you test dependence on dev-libs/libressl:0   It should be slotted

khumarahn commented:
@blueness I added the dependence on libressl:0. Did some more changes as well.

It turned out that libressl does not depend on zlib (while openssl does). There was an issue with compilation with `static` use flag: without openssl, the dependency on zlib[static-libs] was missing. I added an explicit dependence on zlib.

I also moved websocketpp from RDEPEND to DEPEND. (That's a header only library, only used in compilation.)

blueness commented:
pushed
